### DISCUSSION TITLE
Example code to do File search using thread request?

### DISCUSSION THREAD

**Discussion by amavashev** [sashirestela/simple-openai#227] [General]
Example code to do File search using thread request?
Is this the correct snippet?

ThreadMessageRequest.ThreadMessageRequestBuilder request = ThreadMessageRequest.builder()
				.role(ThreadMessageRole.USER);

FileResponse fileResponse = (FileResponse) uploadToVectorStore(fileName);
List<Attachment.AttachmentTool> tools = new ArrayList<>();
tools.add(Attachment.AttachmentTool.FILE_SEARCH); 
List<Attachment> attachments = new ArrayList<>();
attachments.add(new Attachment(fileResponse.getId(), tools));
request.attachments(attachments);

Print out of the thread request when its all setup:

ThreadMessageRequest.ThreadMessageRequestBuilder(role=USER, content=[ContentPart.ContentPartText(text=whats this file?)], attachments=[Attachment(fileId=file-KRjYryEg5wxrKfoUikPb3R, tools=[Attachment.AttachmentTool(type=FILE_SEARCH)])], metadata=null)

**Comment by sashirestela** [comment]
Hi @amavashev. Have you taken a look at this [ConversationV2Demo.java](https://github.com/sashirestela/simple-openai/blob/main/src/demo/java/io/github/sashirestela/openai/demo/ConversationV2Demo.java) code?
